### PR TITLE
Complain if we invite a virtual user to a room on gitter.

### PR DESCRIPTION
**Pull request by Half-Shot**
Can't assign you yet @turt2live :(

### CODE REVIEW

File: lib/BridgedRoom.js [hunk 1]
```diff
@@ -139,81 +143,98 @@ BridgedRoom.prototype.joinAndStart = function() {
...
-    }).then((room) => {
-        this._gitterRoom = room;
+    }).then((room) => this.start(room));
+};
+
+BridgedRoom.prototype.start = function(room) {
+    log.info(`Starting room ${this.gitterRoomName()}`);
+    if(room.subscribe === undefined) {
+        /* This has been fetched from the API but isn't a room object.
+           The nde gitter API for findAll will fetch rooms as json objects,
```
Comment (anoadragon453) [inline lib/BridgedRoom.js]: "nde" -> "node"

File: lib/BridgedRoom.js [hunk 2]
```diff
@@ -139,81 +143,98 @@ BridgedRoom.prototype.joinAndStart = function() {
...
+                (e) => {
+                    endTimer({outcome: "fail"});
 
-            // Wait for all currently-pending send operations to complete
-            //   so we'll have the full set of message IDs
-            Promise.all(this._sendingGuards).then(() => {
-                if (model.id in this._sentMessageIds) {
-                    delete this._sentMessageIds[model.id];
-                    return;
+                    log.warn("Failed: ", e);
```
Comment (anoadragon453) [inline lib/BridgedRoom.js]: Very detailed logging message.

File: lib/Logging.js [hunk 3]
```diff
@@ -38,24 +39,22 @@ class Logging {
             }));
         }
 
-        if(config.files === undefined) {
```
Comment (anoadragon453) [inline lib/Logging.js]: Is this actually a worthwhile change? Was the intention to log no matter what?

File: lib/Main.js [hunk 4]
```diff
@@ -93,6 +92,24 @@ function Main(config) {
                     }
                     endTimer({outcome: "success"});
                     return;
+                } else if (event.type == "m.room.member" && 
+                    bridge.getBot()._isRemoteUser(event.state_key) &&
+                    event.sender !== bridge.getBot().getUserId()) {
+                    // We got an invite but the bot doesn't support them, warn about it.
+                    log.info(`${event.state_key} got invite for ${event.room_id} but we can't do DMs, warning room.`);
+                    const intent = bridge.getIntent(event.state_key);
```
Comment (anoadragon453) [inline lib/Main.js]: This block is a little dense. I'd add 1 or 2 newlines in here somewhere.

File: lib/Main.js [hunk 5]
```diff
@@ -811,8 +836,14 @@ Main.prototype.run = function(port) {
                 }
                 this._bridgedRoomsByMatrixId[matrix_id] = room;
                 this._stateStorage.trackRoom(matrix_id);
-
-                return room.joinAndStart();
+                return promiseAllRooms.then((allRooms) => {
+                    const gitterRoom = allRooms.find((joinedRoom) => joinedRoom.name === remote_id);
+                    if (gitterRoom !== undefined) {
+                        return room.start(gitterRoom);
+                    } else {
```
Comment (anoadragon453) [inline lib/Main.js]: No else needed after return :D

**Comment by Half-Shot** [comment]
I should point out that GH f**ked up and included diffs which aren't on this PR :(. I'll address the ones that are.

File: lib/Main.js (hunk 5 above)
Comment (Half-Shot) [reply lib/Main.js]: [Not in PR]

File: lib/Logging.js (hunk 3 above)
Comment (Half-Shot) [reply lib/Logging.js]: [Not in PR]

File: lib/BridgedRoom.js (hunk 2 above)
Comment (Half-Shot) [reply lib/BridgedRoom.js]: [Not in PR]

File: lib/BridgedRoom.js (hunk 1 above)
Comment (Half-Shot) [reply lib/BridgedRoom.js]: [Not in PR]

**Pull request closed without merging**